src/contrib/cluster/Akka.Cluster.Tools/PublishSubscribe/Internal/Topics.cs

```diff
@@ -329,8 +332,15 @@ protected override bool Business(object message)
     /// </summary>
     internal static class Utils
     {
+        private record MakeKeyInfo(ActorPath Path, string Topic);
+        
         private static readonly System.Text.RegularExpressions.Regex PathRegex = new("^/remote/.+(/user/.+)");
 
+        private static readonly Dictionary<string, string> TopicToEncodedMap = new();
+        private static readonly Dictionary<string, string> EncodedToTopicMap = new();
+        private static readonly Dictionary<MakeKeyInfo, string> MakeKeyMap = new();
+        private static readonly Dictionary<string, MakeKeyInfo> MakeKeyReverseMap = new();
```

Comment:
This strikes me as ever-so-slightly unsafe, in that if there are multiple `ActorSystem`s running in the same app domain, you could have those systems callers potentially mutating a dictionary while another is trying to read...

Reply:
Though let me think about this a bit, this might actually be safe.